src/app/(auth)/signup/components/SignUpChildren.js

```diff
@@ -4,23 +4,46 @@ import {
   FormikDateField,
   FormikSelect,
   FormikTextField,
+  FormikErrors,
+  createHandleSubmit,
 } from "@/app/activities/[activityId]/edit/components/formikFields";
 import { SmFlex } from "@/app/activities/[activityId]/edit/components/responsiveFlexes";
 import { getSchools } from "@/app/api.mjs";
 import AddCircleOutlineIcon from "@mui/icons-material/AddCircleOutline";
 import { Button, Container, MenuItem, Typography } from "@mui/material";
 import dayjs from "dayjs";
 import { Form, Formik } from "formik";
-import { useQuery } from "react-query";
+import { useMutation, useQuery } from "react-query";
+import { useState } from "react"
 import * as Yup from "yup";
 import { SignUpContainer } from "./SignUpAccount";
+import { signupChildren, USER_ID_KEY } from "@/app/api.mjs";
 
 export default function SignUpChildren({ goToNextStep }) {
   const { data: schools } = useQuery("schools", getSchools);
+  const [stayOnPage, setStayOnPage] = useState(false);
+  const [childNumber, setChildNumber] = useState(1);
+  const userId = localStorage.getItem(USER_ID_KEY);
```

Comment:
Why we do not utilize useEffect here and do utilize on another page?

Reply:
I didn't get the error on this page that local storage is not defined. Problem with next.js and SSR [explain](https://sentry.io/answers/referenceerror-localstorage-is-not-defined-in-next-js/). I think, it is a difference between page and a component rendering:

Reply:
Why then do not mark a previous case a client side to make it consistent?

Reply:
I have am error: `localStorage is not defined` despite using the 'use client' directive. This typically happens when localStorage is accessed before the component is mounted on the client side.